utils/utils-lib/src/disk.rs

```diff
@@ -0,0 +1,35 @@
+use url::Url;
+
+/// Normalize the disks if they have a schema, we dont want to change anything
+/// or do any error checking -- the loop will converge to the error state eventually
+pub fn normalize_disk(disk: &str) -> String {
+    let disk = if disk.starts_with("pcie://") {
+        canonical_pcie_address(disk).unwrap_or_else(|| disk.to_string())
+    } else {
+        disk.to_string()
+    };
+
+    Url::parse(&disk).map_or(disk.to_string(), |u| {
+        u.to_file_path()
+            .unwrap_or_else(|_| disk.into())
+            .as_path()
+            .display()
+            .to_string()
+    })
+}
+
+fn canonical_pcie_address(addr: &str) -> Option<String> {
+    let stripped = addr.strip_prefix("pcie://")?;
```

Comment:
Shouldn't we strip the 3 `///` here? otherwise the parts[0] would have a slash or no?

Reply:
Yes, done.